apis/v1alpha1/certificate_authority.go

```diff
@@ -0,0 +1,137 @@
+	// to be disabled.
+	//
+	// The following requirements apply to revocation configurations.
+	//
+	//   - A configuration disabling CRLs or OCSP must contain only the Enabled=False
+	//     parameter, and will fail if other parameters such as CustomCname or ExpirationInDays
+	//     are included.
+	//
+	//   - In a CRL configuration, the S3BucketName parameter must conform to Amazon
+	//     S3 bucket naming rules (https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html).
+	//
+	//   - A configuration containing a custom Canonical Name (CNAME) parameter
+	//     for CRLs or OCSP must conform to RFC2396 (https://www.ietf.org/rfc/rfc2396.txt)
+	//     restrictions on the use of special characters in a CNAME.
+	//
+	//   - In a CRL or OCSP configuration, the value of a CNAME parameter must
+	//     not include a protocol prefix such as "http://" or "https://".
+	//
+	// For more information, see the OcspConfiguration (https://docs.aws.amazon.com/privateca/latest/APIReference/API_OcspConfiguration.html)
+	// and CrlConfiguration (https://docs.aws.amazon.com/privateca/latest/APIReference/API_CrlConfiguration.html)
+	// types.
+	RevocationConfiguration *RevocationConfiguration `json:"revocationConfiguration,omitempty"`
+	// Key-value pairs that will be attached to the new private CA. You can associate
+	// up to 50 tags with a private CA. For information using tags with IAM to manage
+	// permissions, see Controlling Access Using IAM Tags (https://docs.aws.amazon.com/IAM/latest/UserGuide/access_iam-tags.html).
+	Tags []*Tag `json:"tags,omitempty"`
+	// Specifies whether the CA issues general-purpose certificates that typically
+	// require a revocation mechanism, or short-lived certificates that may optionally
+	// omit revocation because they expire quickly. Short-lived certificate validity
+	// is limited to seven days.
+	//
+	// The default value is GENERAL_PURPOSE.
+	UsageMode *string `json:"usageMode,omitempty"`
+}
+
+// CertificateAuthorityStatus defines the observed state of CertificateAuthority
+type CertificateAuthorityStatus struct {
+	// All CRs managed by ACK have a common `Status.ACKResourceMetadata` member
+	// that is used to contain resource sync state, account ownership,
+	// constructed ARN for the resource
+	// +kubebuilder:validation:Optional
+	ACKResourceMetadata *ackv1alpha1.ResourceMetadata `json:"ackResourceMetadata"`
+	// All CRS managed by ACK have a common `Status.Conditions` member that
+	// contains a collection of `ackv1alpha1.Condition` objects that describe
+	// the various terminal states of the CR and its backend AWS service API
+	// resource
+	// +kubebuilder:validation:Optional
+	Conditions []*ackv1alpha1.Condition `json:"conditions"`
+	// The base64 PEM-encoded certificate signing request (CSR) for your private
+	// CA certificate.
+	// +kubebuilder:validation:Optional
+	CertificateSigningRequest *string `json:"certificateSigningRequest,omitempty"`
```

Comment:
Before making a controller release, we need to discuss whether it's safe or not to store a private CA certificate as base64 in the api-server. Maybe we need to extend the controller runtime to start storing these information in [Secret ](https://kubernetes.io/docs/concepts/configuration/secret/)objects instead.

Reply:
Agreed, my team also had this question